Mathlib/LinearAlgebra/Dimension/Finite.lean

```diff
@@ -546,3 +546,12 @@ theorem finrank_le_one (v : M) (h : ∀ w : M, ∃ c : R, c • v = w) : finrank
   · exact (finrank_eq_one v hn h).le
 
 end RankOne
+
+namespace Module
+variable {ι : Type*}
+
+@[simp] lemma finite_finsupp [StrongRankCondition R] : Module.Finite R (ι →₀ R) ↔ Finite ι where
```

Comment:
[Here](https://github.com/leanprover-community/mathlib4/commit/9a14cee38529ee98112f6585212f15b721c84e64) is a generalization to RankCondition. I extracted the proof from #23933 which has a version with Pi instead of Finsupp.

Reply:
I think this PR should be merged first, and then I can reuse a lemma in the above-linked commit in #23933.

Reply:
You can forget about this conversation per my previous comment. I'll PR the commit in #23933 instead.